src/components/Table/TableConfirmButtonsRow.js

```diff
@@ -0,0 +1,146 @@
+    window.addEventListener('resize', this.handleResize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll);
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  saveRowDimensions(element) {
+    if (element) {
+      this.element = element;
+    }
+    if (this.element) {
+      this.setState({
+        rowDimensions: this.element.getBoundingClientRect()
+      });
+    }
+  }
+
+  handleScroll(event) {
+    this.saveRowDimensions();
+  }
+
+  handleResize(event) {
+    this.fetchClientDimensions();
+    this.saveRowDimensions();
+  }
+
+  fetchClientDimensions() {
+    this.setState({
+      window: {
+        width: document.documentElement.clientWidth,
+        height: document.documentElement.clientHeight
+      }
+    });
+  }
+
+  renderConfirmButtons() {
+    const divStyle = this.state.rowDimensions
+      ? this.props.buttonsPosition(this.state.window, this.state.rowDimensions)
+      : {};
+
+    const buttonsClass = this.props.buttonsClass('inline-edit-buttons');
+    return (
+      <div style={divStyle} className={buttonsClass} key="confirmButtons">
+        <ConfirmButton
+          bsStyle="primary"
+          key="confirm"
+          aria-label={this.props.messages.confirmButtonLabel}
+          onMouseUp={() => this.props.onConfirm()}
+        />
+        <CancelButton
+          bsStyle="default"
+          key="cancel"
+          aria-label={this.props.messages.cancelButtonLabel}
+          onMouseUp={() => this.props.onCancel()}
+        />
+      </div>
+    );
+  }
+
+  render() {
+    const editing = this.props.isEditing();
+    const rowClass = editing ? 'editing' : '';
+
+    const elements = [
+      <tr ref={this.saveRowDimensions} className={rowClass} key="tableRow">
+        {this.props.children}
+      </tr>
+    ];
+
+    if (editing && (this.element || this.props.buttonsMountpoint)) {
+      elements.push(
+        // mount the confirm buttons below the table
+        createPortal(
+          this.renderConfirmButtons(),
+          this.props.buttonsMountpoint ||
+            this.element.closest('table').parentNode
+        )
+      );
+    }
+
+    return elements;
+  }
+}
+
+TableConfirmButtonsRow.shouldComponentUpdate = true;
+
+TableConfirmButtonsRow.defaultProps = {
+  isEditing: noop,
+  onConfirm: noop,
+  onCancel: noop,
+  buttonsPosition: () => {},
+  buttonsClass: cls => cls,
+  children: [],
+  messages: {
+    confirmButtonLabel: 'Save',
+    cancelButtonLabel: 'Cancel'
+  },
+  buttonsMountpoint: undefined
+};
+
+TableConfirmButtonsRow.propTypes = {
+  /** Function that determines whether values or edit components should be rendered */
+  isEditing: PropTypes.func,
+  /** Confirm edit callback */
+  onConfirm: PropTypes.func,
+  /** Cancel edit callback */
+  onCancel: PropTypes.func,
+  /** Inject confirm buttons positions */
+  buttonsPosition: PropTypes.func,
+  /** Inject confirm buttons classes */
+  buttonsClass: PropTypes.func,
```

Comment:
Typically we pass the additional classes rather than injecting defaults. Could this be string of classes to add rather than a callback function to inject the default classes?